### ISSUE TITLE
Documentation?

### ISSUE THREAD

**Issue by TechStudent10** [Abb1x/snek#2]
I would really like to make my own OS, but there is no documentation.

So how can I?

**Comment by llamaking136** [comment]
There's really no "one way" to make an OS. You just kinda have to wing it. There are some good tutorials online (like wiki.osdev.org or jamesmolloy.co.uk) but even those OS's are pretty limited. There are a bunch of components that go into an OS, and that's kinda why something like the Linux kernel is still around.

**Comment by TechStudent10** [comment]
As in I want to use Snek.

**Comment by llamaking136** [comment]
Oh, alright. In the Makefile for `pythonOS`, there is a call to `os.system` that includes the arguments to `snek`. It is `snek <input filename> <output filename>`.

**Comment by TechStudent10** [comment]
> Oh, alright. In the Makefile for `pythonOS`, there is a call to `os.system` that includes the arguments to `snek`. It is `snek <input filename> <output filename>`.

Got it. Thanks. Also, I cloned `pythonOS` to my drive and am now making changes. Is there a way I can get user input?

**Comment by llamaking136** [comment]
From they keyboard?

**Comment by TechStudent10** [comment]
Yes

**Comment by llamaking136** [comment]
https://forum.osdev.org/viewtopic.php?f=1&t=28437

**Comment by TechStudent10** [comment]
_In Python_

**Comment by TechStudent10** [comment]
This is the result of me using `input()`.

![image](https://user-images.githubusercontent.com/76978184/150854669-ab5f4cc4-93c0-4504-9a6c-f992fa54c564.png)

**Comment by llamaking136** [comment]
The basic fundamentals still apply. Using Python-to-C translators haven't really been used for OS development before, and `snek` is even newer. If you haven't done OS development before, it'll be even harder doing it in Python.

**Comment by llamaking136** [comment]
> This is the result of me using `input()`.
> 
> ![image](https://user-images.githubusercontent.com/76978184/150854669-ab5f4cc4-93c0-4504-9a6c-f992fa54c564.png)

Well, in that regard, it looks like the bootloader isn't finding the kernel image. Not sure about that.

**Comment by TechStudent10** [comment]
> If you haven't done OS development before, it'll be even harder doing it in Python.

Yeah that's true. I've never actually went through and completed an OS.

> The basic fundamentals still apply. Using Python-to-C translators haven't really been used for OS development before, and `snek` is even newer.

That is something I never knew about. I assumed all the PythonOSes used something like `snek`.

**Comment by TechStudent10** [comment]
> > This is the result of me using `input()`.
> > ![image](https://user-images.githubusercontent.com/76978184/150854669-ab5f4cc4-93c0-4504-9a6c-f992fa54c564.png)
> 
> Well, in that regard, it looks like the bootloader isn't finding the kernel image. Not sure about that.

I think that `input()` isn't being recognized in C, causing it to not compile correctly and just continue and error out.

**Comment by llamaking136** [comment]
Do you have a repo for your code?

**Comment by TechStudent10** [comment]
Nah, but I can create one quickly.

**Comment by llamaking136** [comment]
Okay, I might be able to diagnose the error then.

**Comment by TechStudent10** [comment]
Okay it's [`TechStudent11/snek-test`](https://github.com/TechStudent11/snek-test)

**Comment by llamaking136** [comment]
Okay, thanks.

**Comment by llamaking136** [comment]
Alright. I see the issue. The reason that Limine can't find the kernel executable is because the compiler isn't happy. I re-wrote the build script to make sure that every call to `os.system` recognized an error and stopped the program. It can't find `input` because it isn't in the OS (yet).

**Comment by llamaking136** [comment]
So you'll have to do it the old-fashioned way and read wiki.osdev.org.

**Comment by TechStudent10** [comment]
> Alright. I see the issue. The reason that Limine can't find the kernel executable is because the compiler isn't happy. I re-wrote the build script to make sure that every call to `os.system` recognized an error and stopped the program. It can't find `input` because it isn't in the OS (yet).

That makes sense.

> So you'll have to do it the old-fashioned way and read wiki.osdev.org.

Ah I guess. When I saw Snek I got super excited that now I was able to write an entire OS in my favorite language but oh well C and Assembly it is.

**Comment by llamaking136** [comment]
It's fine if you do it with `snek`, but I'd recommend C + Assembly to understand the wiki and for usability. You know what, I should make a C-to-Python translator instead...

**Comment by TechStudent10** [comment]
Thanks for your help.

**Comment by llamaking136** [comment]
Yeah, no problem.

**Comment by TechStudent10** [comment]
> You know what, I should make a C-to-Python translator instead...

That would be awesome. I've never seen a C to Python translator. Maybe instead of CPython, you'll call it PythonC haha.

**Comment by JTSJohnny** [comment]
Try using assembly.
